app/controllers/email_alert_subscriptions_controller.rb

```diff
@@ -35,11 +46,33 @@ def at_least_one_filter_chosen?
   end
 
   def chosen_options
-    params.permit("filter" => {})['filter'].to_h
+    options = params.permit("filter" => {})['filter'].to_h
+    return options unless params[:hidden_params].present?
+
+    options.merge(params[:hidden_params].to_unsafe_hash)
```

Comment:
Can you explain the need to make the params unsafe? I'm wondering if there is a way to rework this to leverage Rails' strong parameters.